js/web/lib/wasm/jsep/webgpu/ops/batch-norm.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {env} from 'onnxruntime-common';
+
+import {TensorView} from '../../tensor-view';
+import {ShapeUtil} from '../../util';
+import {AttributeWithCacheKey, createAttributeWithCacheKey} from '../attribute-with-cache-key';
+import {ComputeContext, ProgramInfo} from '../types';
+
+import {createTensorShapeVariables, enableShapesUniforms, getMaxComponents, inputVariable, outputVariable, ShaderHelper} from './common';
+
+export interface BatchNormAttributes extends AttributeWithCacheKey {
+  readonly epsilon: number;
+  readonly momentum: number;
+  readonly spatial: boolean;
+  readonly trainingMode: boolean;
+  readonly format: 'nhwc'|'nchw';
+  readonly outputCount: number;
+}
+
+const validateInputs = (inputs: readonly TensorView[], attributes: BatchNormAttributes): void => {
+  if (!inputs || inputs.length !== 5) {
+    throw new Error('BatchNormalization requires 5 inputs');
+  }
+
+  const checkShapeEqual = (actual: readonly number[], expected: readonly number[], message: string) => {
+    const r = expected.length;
+    if (r !== actual.length) {
+      throw new Error(`${message}: num dimensions != ${r}`);
+    }
+    expected.forEach((v, i) => {
+      if (v !== actual[i]) {
+        switch (i % 100) {
+          case 11:
+          case 12:
+          case 13:
+            throw new Error(`${message}: ${i}th dimension != ${v}`);
+          default:
+        }
+        switch (i % 10) {
+          case 1:
+            throw new Error(`${message}: ${i}st dimension != ${v}`);
+          case 2:
+            throw new Error(`${message}: ${i}nd dimension != ${v}`);
+          case 3:
+            throw new Error(`${message}: ${i}rd dimension != ${v}`);
+          default:
+            throw new Error(`${message}: ${i}th dimension != ${v}`);
+        }
```

Comment:
perhaps make this simpler... something like `${message}: dim[${i}] do not match` is good enough

Reply:
Done